ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -78,27 +159,29 @@ fi
 # Get desired compute node count
 getDesiredComputeCount
 
-# wait for all nodes to reach Ready=true to ensure that all machines and nodes came up, before we run
-# any e2e tests that might require specific workload capacity.
-echo "$(date) - waiting for all nodes to be ready in $READY_WAIT_TIMEOUT"
+# Check if this is a HCP cluster
+is_hcp_cluster="$(rosa describe cluster -c "$CLUSTER_ID" -o json  | jq -r ".hypershift.enabled")"
+log "hypershift.enabled is set to $is_hcp_cluster"
+
 ret=0
-oc wait nodes --all --for=condition=Ready=true --timeout="$READY_WAIT_TIMEOUT" || ret=$?
+echo "Wait for all nodes to be ready and schedulable."
+waitForReady "$desired_compute_count" || ret=$?
 
 if [[ "$ret" == 0 ]]; then
     # Get count of worker only nodes in Ready state
-    node_count="$(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/SchedulingDisabled!=,node-role.kubernetes.io/infra!= --output jsonpath="{.items[?(@.status.conditions[-1].type=='Ready')].status.conditions[-1].type}" | wc -w | xargs)"
```

Comment:
think we should keep the scheduling disabled here, need all nodes to be ready and scheduling enabled

Reply:
I deleted it because I was thinking that it's actually not a role. It sure might have a label but filtering by role for SchedulingDisabled seemed to being doing nothing. Let me know if otherwise.

Reply:
gotcha!! yeh I dont think its doign anything. good to take out

```
 % oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/SchedulingDisabled!= 
ip-10-0-142-193.us-east-2.compute.internal   Ready,SchedulingDisabled   worker   175m   v1.27.3+4aaeaec
ip-10-0-168-92.us-east-2.compute.internal    Ready                      worker   177m   v1.27.3+4aaeaec
ip-10-0-201-99.us-east-2.compute.internal    Ready                      worker   179m   v1.27.3+4aaeaec
```